src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1170,4 +1172,124 @@ private function addRobotsIndexSection(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addHttpClientSection(ArrayNodeDefinition $rootNode)
+    {
+        $subNode = $rootNode
+            ->children()
+                ->arrayNode('http_client')
+                    ->info('HTTP Client configuration')
+                    ->canBeEnabled()
+                    ->fixXmlConfig('client')
+                    ->children();
+
+        $this->addHttpClientOptionsSection($subNode);
+
+        $subNode = $subNode
+                        ->arrayNode('clients')
+                            ->useAttributeAsKey('name')
+                            ->normalizeKeys(false)
+                            ->arrayPrototype()
+                                ->children();
+
+        $this->addHttpClientOptionsSection($subNode);
+
+        $subNode = $subNode
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    private function addHttpClientOptionsSection(NodeBuilder $rootNode)
+    {
+        $rootNode
+            ->integerNode('max_host_connections')
+                ->info('The maximum number of connections to a single host.')
+            ->end()
+            ->arrayNode('default_options')
```

Comment:
These are the default options for each request that's made. What about `default_request_options` (it's even documented as `Default requests' options` in `HttpClient` phpdoc). And also, should we make these options *not* automatically be used as the "default options" for all the other HTTP clients?

The problem is that there are 2 concepts conflicting right now. Does "default" mean:
A) default options for the request that can be overridden on each request
OR
B) default options that will be applied to all other HTTP clients

Right now, it's both - that confused me. It's really (A), so maybe we should make it *only* (A) and clarify the name.

Reply:
right now, it is default options applied to all other HTTP clients for their default request options...

Reply:
Correct, it's A and B. If one want to configure the default request options for the default client only, one should set them under `framework.http_cllient.clients.http_client.default_options` (since `http_client` is the name of the service that is aliased for `HttpClientInterface`).
I'm not sure I understand how your suggestion would apply. To me, all is fine :)

Reply:
I was suggesting that we remove (B), but it's subjective. The question is: is it valuable for these options to cascade onto other clients? Or is it annoying/surprising, and then I am unsetting options on my individual clients, because I don't want them to cascade from my default client?

I'm not convinced that re-using options across clients will be all that common - things like auth, base_uri, headers all seem *very* client-specific. I could argue that it might be surprising if my auth on my default client starts getting sent to another server via another client (because I didn't realize the options would cascade down to my client).

Reply:
e.g. cabundle/capath: you want to configure them once for all

Reply:
Hmm, ok, really fine by me then.